Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Templates/LambdaFunctionTemplate.tt

```diff
@@ -476,7 +486,7 @@ namespace <#= _model.LambdaMethod.ContainingNamespace #>
         else
         {
 #>
-            return <#= _model.LambdaMethod.IsAsync ? "await " : "" #><#= _model.LambdaMethod.ContainingType.Name.ToCamelCase() #>.<#= _model.LambdaMethod.Name #>(<#= parameters #>);
+            return <#= _model.LambdaMethod.ReturnsGenericTask ? "await " : "" #><#= _model.LambdaMethod.ContainingType.Name.ToCamelCase() #>.<#= _model.LambdaMethod.Name #>(<#= parameters #>);
```

Comment:
I think this is not supported, since you have to use `IHttpResults` with API Gateway (right?)
```       
        [LambdaFunction()]
        public IHttpResults GetOrderInfo()
        {
            return HttpResults.Ok(new OrderInfo { OrderId = "132", CustomerId = "999" });
        }
```

But it results in this compile error, which I think may be confusing because the `Stream` is hidden from the user's code. Can we give a better error for if `HttpResults` are returned outside of API Gateway?
```
Cannot implicitly convert type 'Amazon.Lambda.Annotations.APIGateway.IHttpResults' to 'System.IO.Stream'. An explicit conversion exists (are you missing a cast?)
```

Reply:
I added a new Diagnostic compiler error message that will give a more meaningful error message when the `IHttpResult` is attempted to be used as a return type for a non API Lambda function.

I also updated the generator to not run the CloudFormation sync if our compiler error diagnostics are emitted. No sense mutating the CF template if we now we have a bad build.